nadeo_event_api/src/nadeo_event_api/api/structure/settings/script_settings.py

```diff
@@ -2,17 +2,42 @@
 
 from ...structure.enums import RespawnBehavior
 
+# Nadeo api documentation: https://wiki.trackmania.io/en/dedicated-server/Usage/OfficialGameModesSettings
 
-class ScriptSettings(ABC):
+# 
+class BaseScriptSettings():
     def __init__(
-        self,
+        self,         
         chat_time: int = 10,
         force_laps_number: int = -1,
         respawn_behavior: RespawnBehavior = RespawnBehavior.DEFAULT,
         warmup_duration: int = 0,
         warmup_number: int = 0,
         warmup_timeout: int = -1,
         pick_ban_enable: bool = False,
+                 ):
+        
+        self._chat_time = chat_time
+        self._force_laps_number = force_laps_number
+        self._respawn_behavior = respawn_behavior
+        self._warmup_duration = warmup_duration
+        self._warmup_number = warmup_number
+        self._warmup_timeout = warmup_timeout
+        self._pick_ban_enable = pick_ban_enable
+
+
+class ScriptSettings(ABC):
+    def __init__(
+        self,
+        
+        base_script_settings_variables: BaseScriptSettings
```

Comment:
This should be able to take a default BaseScriptSettings, and I also think the naming can be changed. I suggest the line:
`base_script_settings: BaseScriptSettings = BaseScriptSettings()`